ElectricalCircuitSimulator/SimulationEngine/src/Matrix.cpp

```diff
@@ -20,27 +19,44 @@ namespace SimulationEngine {
 
         m_iNumRows = iRows;
         m_iNumColumns = iColumns;
-        m_oMatrix = new double*[m_iNumRows];
+        m_pMatrix = new double*[m_iNumRows];
         for(iRow = 0; iRow < m_iNumRows; iRow++) {
-            m_oMatrix[iRow] = new double[m_iNumColumns];
+            m_pMatrix[iRow] = new double[m_iNumColumns];
         }
         clear();
     }
 
+    Matrix::Matrix(const Matrix& oMatrix) {
+        int iRow;
+        int iColumn;
+        m_iNumRows = oMatrix.getNumRows();
+        m_iNumColumns = oMatrix.getNumColumns();
+        m_pMatrix = new double* [m_iNumRows];
+        for (iRow = 0; iRow < m_iNumRows; iRow++) {
+            m_pMatrix[iRow] = new double[m_iNumColumns];
+        }
+
+        for (iRow = 0; iRow < m_iNumRows; iRow++) {
+            for (iColumn = 0; iColumn < m_iNumColumns; iColumn++) {
+                m_pMatrix[iRow][iColumn] = oMatrix(iRow, iColumn);
+            }
+        }
+    }
+
     double Matrix::getValue(const int iRow, int const iColumn) const {
         if ((iRow >= m_iNumRows) || (iColumn >= m_iNumColumns)) {
             cout << "Location beyond dimensions of matrix!" << endl;
             throw invalid_argument("Location beyond dimensions of matrix!");
         }
-        return m_oMatrix[iRow][iColumn];
+        return m_pMatrix[iRow][iColumn];
     }
 
     void Matrix::setValue(const int iRow, const int iColumn, const double dValue) {
```

Comment:
do you want to get rid of the getValue function now that you have the () operator?